src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/EJSONWrapper.java

```diff
@@ -118,4 +121,46 @@ private static byte[] bytesFromByteBuffer(ByteBuffer buffer) {
     buffer.get(bytes);
     return bytes;
   }
+
+  // Returns the float array value from the binary EJSON wrapper
+  public float[] getVectorValueForBinary() {
+    if (type != EJSONType.BINARY) {
+      throw new RuntimeException("Vector value can only be extracted from binary EJSON wrapper");
+    }
+    return toFloatArray(value.textValue());
+  }
+
+  // Utility method for converting binary wrapped value to float array
+  public static float[] toFloatArray(String base64Vector) {
+    byte[] binaryPayload;
+    try {
+      binaryPayload = Base64Util.decodeFromMimeBase64(base64Vector);
+    } catch (IllegalArgumentException e) {
+      throw new RuntimeException(
+          "Invalid content in EJSON $binary wrapper: not valid Base64-encoded String, problem: %s"
+              .formatted(e.getMessage()));
+    }
+
+    try {
+      return CqlVectorUtil.bytesToFloats(binaryPayload);
+    } catch (IllegalArgumentException e) {
+      throw new RuntimeException(
+          "Invalid content in EJSON $binary wrapper, problem: %s".formatted(e.getMessage()));
+    }
+  }
+
+  // Utility method for converting float array to binary vector value
+  public static String binaryFormatString(float[] vector) {
+    ByteBuffer byteBuffer = ByteBuffer.allocate(vector.length * 4); // 4 bytes per float
+
+    for (float val : vector) {
```

Comment:
Fix before merge: the code in CqlVectorUtil.floatsToBytes can be used to turn the float[] into byte[] you can use that without needing to go to byte buffer.

OK to fix after merge:
This looks to duplicate some of the features in Base64Util.encodeAsMimeBase64, the last line is the same as in that function. 

 work with @tatu on how to refactor , this feels like it should be a function on Base64Util (it can be called from here but should be there)

Reply:
Make sense was just refactoring what was used in Collections. Changed to use method `CqlVectorUtil.floatsToBytes()`. @tatu-at-datastax Should we create a different util class for handling binary format, I don't think it should use `Cql`?